micro-rdk/src/common/registry.rs

```diff
@@ -202,6 +209,18 @@ impl ComponentRegistry {
             dependencies: dependency_func_map,
         }
     }
+    pub fn register_button(
+        &mut self,
+        model: impl Into<String>,
+        constructor: &'static ButtonConstructor,
+    ) -> Result<(), RegistryError> {
+        let model = model.into();
+        if self.buttons.contains_key(&model) {
+            return Err(RegistryError::ModelAlreadyRegistered(model));
+        }
+        let _ = self.buttons.insert(model, constructor);
```

Comment:
it returns an option if there was a value associated with that key above. we check if there is already a value above.